src/main/java/com/minecolonies/coremod/entity/pathfinding/MinecoloniesAdvancedPathNavigate.java

```diff
@@ -532,18 +671,50 @@ protected void pathFollow()
 
     /**
      * Don't let vanilla rapidly discard paths, set a timeout before its allowed to use stuck.
-     *
-     * @param positionVec3
      */
     @Override
-    protected void checkForStuck(Vec3d positionVec3)
+    protected void checkForStuck(@NotNull final Vec3d positionVec3)
     {
         if (world.getGameTime() - pathStartTime < MIN_KEEP_TIME)
         {
             return;
         }
 
-        super.checkForStuck(positionVec3);
```

Comment:
think clearpath could use some dismount when it somehow gets stuck within the minecart(make them timed lifetime aswell?)

Reply:
they will automatically dismount when they try a new path that doesn't take the rail.